pkg/sql/schemachanger/scexec/exec_backfill.go

```diff
@@ -324,7 +326,7 @@ func runBackfiller(
 	im := deps.IndexMerger()
 	const op = "run backfills and merges"
 	bf := func(ctx context.Context, p *BackfillProgress) error {
-		return runBackfill(ctx, deps.IndexSpanSplitter(), ib, *p, tracker, tables[p.TableID])
+		return runBackfill(ctx, deps.IndexSpanSplitter(), ib, *p, tracker, job, tables[p.TableID])
 	}
 	mf := func(ctx context.Context, p *MergeProgress) error {
 		return im.MergeIndexes(ctx, *p, tracker, tables[p.TableID])
```

Comment:
Should we also plumb the job in here, to be passed on to a `TryToProtectBeforeGC` call? At this point in this branch we have these calls for all validations and for backfills but not for merges, and merges don't seem special to me.

Reply:
Yes good point we should cover all cases with a AOST queries